### ISSUE TITLE
Android build

### ISSUE THREAD

**Issue by Xandev** [RedpointGames/Protogame#213]
Hi,   
What are the steps to build and deploy working game based on Protogame?

I'm asking because after genereting empty project for android i've got:
`'SystemUiFlags' does not contain a definition for 'ImmersiveSticky'` in GameActivity.cs. 
When i delete `| SystemUiFlags.ImmersiveSticky`, there're more errors which are informing about missing `@drawable/icon, @string/app_name and @style/Protogame.Splash`.
After adding what is missing I can build and deploy apk, but there is black screen and nothing more.
Shouldn't be there pink screen with fps text etc.?

Am I missing something?

Regards

**Comment by hach-que** [comment]
Uh, it should just work straight out of the box.

What version of the Android SDK do you have and what version of Xamarin do you have?

**Comment by Xandev** [comment]
Somehow project was set to Android 4.2. After setting it to 4.4+ error with ImmersiveSticky is gone.
However there's still problem with missing drawable and values.

What exactly do you mean by Android SDK version? 
Tools: 25.2.5
Platform-tools: 26
APIs installed: 23, 22, 21, 19, 17

Xamarin for VisualStudio: 4.6.0.299
Xamarin.Android.SDK: 7.4.0.21

**Comment by hach-que** [comment]
What Visual Studio? 2017 or 2015?

Trying to replicate the issue now from the platformer template.

**Comment by Xandev** [comment]
2017 Community. 
Still drawable and values are missing.
Btw. I'm generating project via protobuild GUI for windows.

**Comment by hach-que** [comment]
I just pushed a fix for the Protobuild Manager so it'll properly open up solutions in VS2017 (this was blocking me testing a new Android project since you can't use Xamarin in 2015 any more).

Once that's out I'll test the Android build from the platformer template.

**Comment by hach-que** [comment]
Okay, so yeah the templates should include these missing resources.

We updated Protogame a while back on Android to support asynchronous app loading (it makes the game load experience much nicer for the player). Unfortunately we didn't update the templates to contain starting files for these resources, which asynchronous app loading depends on.

I'm working on getting starting files into the template repos now so that it should work again.

**Comment by hach-que** [comment]
I have a PR lined up for the blank template that should at least fix the compilation and loading resources issues https://github.com/RedpointGames/Protogame.Template.Blank/pull/68.

I've replicated the black screen issue on the blank template.

**Comment by hach-que** [comment]
I've figured out what's going wrong, but I can tell it's going to take some time to fix the underlying issue.

In the meantime, here's a workaround that will get things working for you:

- Open up a command prompt in the "<your project>\Protogame" directory after you generate the solution for the first time.
- Run `.\Protobuild.exe --swap-to-source "https-nuget-
v3://api.nuget.org/v3/index.json|Protoinject"`
- Then run `.\Protobuild.exe --generate Windows,Android`
- Inside Visual Studio, open the Android solution
- For `Protoinject.FactoryGenerator.Windows`, `ProtogamePostLibBuild.Windows` and `ProtogamePostGameBuild.Windows`, right-click on them and click "Build"
- Now you can build and deploy your Android game, and it should launch successfully

![screenshot_20170827-043232](https://user-images.githubusercontent.com/504826/29744096-e88dffda-8ae0-11e7-8853-109529a07aca.png)

---

What's happening here is that when Protoinject is installed as a binary package, the post-build step inside Android projects has the following:

```
<Message Importance="high" Text="Running &quot;Protoinject.FactoryGenerator&quot; post-build hook..." Condition="Exists('C:\Users\jrhod\Documents\Projects\Protogame.Template.Blank\Protogame\Protoinject\Android\Protoinject.FactoryGenerator\AnyCPU\Protoinject.FactoryGenerator.exe')" />
    <Exec Condition="Exists('C:\Users\jrhod\Documents\Projects\Protogame.Template.Blank\Protogame\Protoinject\Android\Protoinject.FactoryGenerator\AnyCPU\Protoinject.FactoryGenerator.exe')" WorkingDirectory="C:\Users\jrhod\Documents\Projects\Protogame.Template.Blank\PROJECT_NAME" Command="&quot;C:\Users\jrhod\Documents\Projects\Protogame.Template.Blank\Protogame\Protoinject\Android\Protoinject.FactoryGenerator\AnyCPU\Protoinject.FactoryGenerator.exe&quot; &quot;@(IntermediateAssembly)&quot; &quot;@(ReferencePath)&quot;" />
```

This is because it's looking for the factory generator inside the Android package, and the Android package doesn't ship it because it's an executable for the host platform. It *does* exist inside the `Windows` package which is available next to it (because Protogame has host tools in it).

However, this is indicative of a larger problem. If someone installs a package for Android and the parent solution isn't using host executables already, and the package contains a post-build step, then Protobuild won't do the right thing. It'll set up the hook to point inside the non-desktop package, and the desktop/host package that contains it won't even exist.

What might need to be done here is add a flag to binary packages that indicates that the host platform version of the package should *also* be installed when packages are resolved. Then when Protobuild resolves a package, it checks for the existence of this flag - if it exists, it queues the host-platform version of the package also for download and makes sure it's unpacked.

Once this is done, there needs to be some way of indicating that a post-build hook should use the host platform instead of the current platform. There isn't anything like this at all in Protobuild at the moment (redirecting from one binary package's platform to another for a project), but it might be okay if it's hard-coded logic just for resolution of post-build steps?
